src/index.ts

```diff
@@ -169,10 +186,41 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
                 targetPath = path.relative(path.dirname(htmlFileDirectory), filepath)
             }
             const ext = path.parse(filepath).ext
+
+            // Inline the JavaScript and CSS files if the option is set.
+            const { inline } = htmlFileConfiguration
+
+            const isInline = (inline_: HtmlFileConfiguration['inline'] | undefined, ext_: '.css' | '.js') => {
+                if (!inline_) {
+                    return false
+                }
+                const extension = ext_.replace('.', '') as 'css' | 'js'
+                return (
+                    (typeof inline_ === 'boolean' && inline_ === true) ||
+                    (typeof inline_ === 'object' && inline_[extension] === true)
+                )
+            }
+
             if (ext === '.js') {
                 const scriptTag = document.createElement('script')
-                scriptTag.setAttribute('src', targetPath)
+                // Check if the JavaScript should be inlined.
+                if (isInline(inline, ext)) {
+                    console.log('Inlining script', filepath)
```

Comment:
TODO: We use `logInfo` if we want to do verbose logging: 
```suggestion
                    logInfo && console.log('Inlining script', filepath)
```